packages/core/src/providers/vipps.ts

```diff
@@ -0,0 +1,90 @@
+import type { OIDCConfig, OIDCUserConfig } from "./index.js"
+
+/** @see [User Profile Structure](https://vippsas.github.io/vipps-developer-docs/docs/APIs/login-api/vipps-login-api#userinfo) */
+export interface VippsProfile extends Record<string, any> {
+  sid: string
+  birthdate: string
+  email: string
+  email_verified: boolean
+  family_name: string
+  given_name: string
+  name: string
+  phone_number: string
+  sub: string
+}
+
+interface AdditonalConfig {
+  redirectUri?: string
+}
+
+/**
+ * @see [Vipps Login API](https://vippsas.github.io/vipps-developer-docs/docs/APIs/login-api/vipps-login-api)
+ *
+ * ## Example
+ *
+ * ```ts
+ * import Vipps from "@auth/core/providers/vipps"
+ * ...
+ * providers: [
+ *  Vipps({
+ *    clientId: process.env.VIPPS_CLIENT_ID,
+ *    clientSecret: process.env.VIPPS_CLIENT_SECRET,
+ *  })
+ * ]
+ * ...
+ * ```
+ * ::: note
+ * If you're testing, make sure to override the issuer option with apitest.vipps.no
+ * :::
+ */
+
+export default function Vipps<P extends VippsProfile>(
+  options: OIDCUserConfig<P> & AdditonalConfig
+): OIDCConfig<P> {
+  options.issuer ??= "https://api.vipps.no/access-management-1.0/access"
+  return {
+    id: "vipps",
+    name: "Vipps",
+    type: "oidc",
+    client: {
+      token_endpoint_auth_method: "client_secret_post",
+    },
+    authorization: {
```

Comment:
Since this is an OIDC provider can we just use the discovery endpoint? https://vippsas.github.io/vipps-developer-docs/docs/APIs/login-api/vipps-login-api#manual-integration

Reply:
Although, from what I can tell, we do need the `issuer` and `token_endpoint_auth_method` anyways.